File: testing/cppsrc/binarystep.cpp
#include <iostream>
#include <vector>
#include <algorithm>
#include <cmath>

using namespace std;

class binarystep {
	public:	
		vector <double> forward(vector <double> input) {
      transform(input.begin(), input.end(), input.begin(), [](double input){ return (input < 0) ? 0 : 1; });
      return input;
		}

		vector <double> backward(vector <double> input) {
			return { [1] * input.size() };
		}
};



int main() {
	binarystep activation;
	double input = 0.3;
	double outputforward = activation.forward(input);
	double outputbackward = activation.backward(input);

	cout << "forward: " << outputforward << "\n";
	cout << "backward: " << outputbackward << "\n";

	return 0;
}

File: aiinpy/binarystep.py
import numpy as np

class binarystep:
  def forward(self, input):
    equation = np.vectorize(self.equationForbinaryStep, otypes=[float])
    return equation(input)

  def equationForbinaryStep(self, input):
    return 0 if (input < 0) else 1
    
  def backward(self, input):
    return 1